src/utils.ts

```diff
@@ -1158,3 +1158,23 @@ export function hasAtomicOperators(doc: Document | Document[]): boolean {
   const keys = Object.keys(doc);
   return keys.length > 0 && keys[0][0] === '$';
 }
+
+export function deepFreeze<O extends Record<string, any>>(object: O): Readonly<O> {
+  // Retrieve the property names defined on object
+  const propNames = Object.getOwnPropertyNames(object);
+
+  // Freeze properties before freezing self
+
+  for (const name of propNames) {
+    const value = object[name];
+
+    if (name === 'session') continue;
+    if (ArrayBuffer.isView(value)) continue;
```

Comment:
This is pretty lame skip logic, but I'm thinking maybe deepFreezing is overkill anyway. thoughts?

Reply:
I think it would be useful to explain why those need to be skipped with comments. Is there a reason we can't use a popular public domain implementation of this like the [deep-freeze](https://github.com/substack/deep-freeze/blob/master/index.js) npm module?

Reply:
I got the code from the same place that dep gets it from the MDN docs, I figure for the small changes we might want to make, might as well in house it.
I did document these skips but I think I can come up with better logic.